### PR TITLE
Fixing RAM Init generation due to miss-orientation of params (reversing)

### PR DESCRIPTION
**Pull request by manadher**

### CODE REVIEW

File: gatelevel_readers/reconstruct_verilog/src/reconstruct_ram18kx2.h [hunk 1]
```diff
@@ -340,12 +340,22 @@ struct TDP_RAM18KX2_instance {
     return sram;
   }
 
-  std::string get_init_i1(const std::string &init1, const std::string &parity1,
```
Comment (alaindargelas) [inline gatelevel_readers/reconstruct_verilog/src/reconstruct_ram18kx2.h:343]: The same fix has to happen in the other file:
https://github.com/os-fpga/Raptor_Tools/blob/74f2ab79214015eaa8b6150f066dfe0dd7485b92/gatelevel_readers/reconstruct_verilog/src/reconstruct_ram36k.h#L372

File: gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/fullSizeTest/main.cpp [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+  // Generate the random string
+  std::string result;
+  for (int i = 0; i < length; ++i) {
+    result += std::to_string(distribution(gen));
+  }
+
+  return result;
+}
+std::string extract_sram1(const std::string &init, const std::string &parity) {
```
Comment (alaindargelas) [inline gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/fullSizeTest/main.cpp]: Why is the test creating a copy of the code?
If the actual code changes, this test will not detect the breakage.
Simply include the actual header and test the actual code (https://github.com/os-fpga/Raptor_Tools/blob/74f2ab79214015eaa8b6150f066dfe0dd7485b92/gatelevel_readers/reconstruct_verilog/src/reconstruct_ram18kx2.h#L343), not a copy of the code.

File: gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/smallSizeTest/run.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+mkdir build
```
Comment (alaindargelas) [inline gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/smallSizeTest/run.sh]: This run.sh is not a standard entry point, it should be called from a Makefile.
When stepping in this directory, typing "make" or "make test" should call this script. The entire Raptor source tree uses "make" as entry points in each subsystem. Let's not deviate from that practice.

Once the Makefile is created, then invoke it (make or make test) from the Github Actions, so the test is actually run automatically when checking in code:
https://github.com/os-fpga/Raptor_Tools/blob/main/.github/workflows/gate_level_readers.yml

**Review by alaindargelas — CHANGES_REQUESTED** [review]

File: gatelevel_readers/reconstruct_verilog/src/reconstruct_ram18kx2.h (hunk 1 above)
Comment (manadher) [reply gatelevel_readers/reconstruct_verilog/src/reconstruct_ram18kx2.h:343]: Alain, work was not marked done, as I mentioned in the JIRA case the RAM36 has yet to be addressed, it's just that it was late/early in this morning and I wanted to speed testing with Integration tests (RAM18 tests would pass not RAM36)

File: gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/fullSizeTest/main.cpp (hunk 2 above)
Comment (manadher) [reply gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/fullSizeTest/main.cpp]: Not a finished work yet, in fact The code in this main was copied back to the .h file as I was making the fix here ...
By the time I make the final check-in things will straighten up.

File: gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/smallSizeTest/run.sh (hunk 3 above)
Comment (manadher) [reply gatelevel_readers/reconstruct_verilog/tests/RAM/RAM18Kx2_MEM_CHECK/smallSizeTest/run.sh]: I will.

**Pull request merged**